### PR TITLE
dev-qt/qtnetwork: add USE=libressl and patch for LibreSSL support

### PR DESCRIPTION
**Pull request by steils**
Closes: https://bugs.gentoo.org/562050
Package-Manager: Portage-2.3.86, Repoman-2.3.20
Signed-off-by: Stefan Strogin <steils@gentoo.org>

### CODE REVIEW

**Comment by a17r** [comment]
Is it upstream now?

**Comment by steils** [comment]
> Is it upstream now?

No, it's not upstream. This patch is from libressl overlay. It has been maintained there for more than a year (and I think quite successfully). And I intend to continue maintaining in gentoo.git if will be accepted.
After all this time and LibreSSL development the patch has become relatively small and easy to maintain. I don't think there will be any trouble with it.

File: dev-qt/qtnetwork/qtnetwork-5.14.1.ebuild [hunk 1]
```diff
@@ -20,13 +20,18 @@ DEPEND="
...
+		!libressl? ( dev-libs/openssl:0=[bindist=] )
+		libressl? ( dev-libs/libressl:0= )
+	)
 "
 RDEPEND="${DEPEND}
 	connman? ( net-misc/connman )
 	networkmanager? ( net-misc/networkmanager )
 "
 
+PATCHES=( "${FILESDIR}"/${P}-libressl.patch )
```
Comment (a17r) [inline dev-qt/qtnetwork/qtnetwork-5.14.1.ebuild]: missing bug reference, should have inline comment `# Not upstreamable`

File: dev-qt/qtnetwork/files/qtnetwork-5.14.1-libressl.patch [hunk 2]
```diff
@@ -0,0 +1,106 @@
+diff --git a/src/network/configure.json b/src/network/configure.json
```
Comment (a17r) [inline dev-qt/qtnetwork/files/qtnetwork-5.14.1-libressl.patch]: Missing patch header, preferably `git format-patch --stdout` style.

File: dev-qt/qtnetwork/qtnetwork-5.14.1.ebuild (hunk 1 above)
Comment (steils) [reply dev-qt/qtnetwork/qtnetwork-5.14.1.ebuild]: Thanks. Fixed.

File: dev-qt/qtnetwork/files/qtnetwork-5.14.1-libressl.patch (hunk 2 above)
Comment (steils) [reply dev-qt/qtnetwork/files/qtnetwork-5.14.1-libressl.patch]: Thanks. Fixed.

**Comment by steils** [comment]
Updated the PR for qtnetwork-5.15.0.

**Comment by Pesa** [comment]
Yes, sorry, now that 5.15 is out I'm ok with accepting the patch.

**Comment by steils** [comment]
@Pesa Thank you!
So, you haven't yet hit the approve button... Shall I wait for you to push the commit? Or somebody else from Qt project to approve it?

**Comment by Pesa** [comment]
What do others think? @a17r @Chiitoo

**Comment by a17r** [comment]
I'm fine with it if you are ;)

**Comment by steils** [comment]
Closed in https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=568a924b3f62d0acff635b2379aedd85ebbc0b99

**Pull request closed without merging**